### PR TITLE
Ben/tables and deberta

### PR DESCRIPTION
**Pull request by benleetownsend**
Still waiting for the Deberta base model tests to run. But otherwise a clean run of passing tests.

### CODE REVIEW

**Comment by benleetownsend** [comment]
Supersedes #744 and #746

File: tests/test_classifier.py [hunk 1]
```diff
@@ -371,7 +371,7 @@ def test_generate_text_stop_early(self):
         # A dirty mock to make all model inferences output a hundred _classify_ tokens
         fake_estimator = MagicMock()
         model.get_estimator = lambda *args, **kwargs: (fake_estimator, [])
-        model.input_pipeline.text_encoder._lazy_init()
+        #model.input_pipeline.text_encoder._lazy_init()
```
Comment (benleetownsend) [inline tests/test_classifier.py:374]: remove.

**Comment by benleetownsend** [comment]
Full test run has been done and passed.

File: finetune/base_models/bert/modeling.py [hunk 2]
```diff
@@ -612,11 +647,9 @@ def embedding_postprocessor(
...
             if anneal_reading_order:
                 full_position_embeddings *= tf.cast(
-                    tf.random.uniform(
-                        shape=tf.shape(input=full_position_embeddings
-                        )
-                    ) > reading_order_decay_rate,
-                    tf.float32
+                    tf.random.uniform(shape=tf.shape(input=full_position_embeddings))
+                    > reading_order_decay_rate,
+                    tf.float32,
```
Comment (madisonmay) [inline finetune/base_models/bert/modeling.py]: Do we need tf.float32 cast or could we pull dtype from some other tensor

File: finetune/base_models/bert/modeling.py [hunk 3]
```diff
@@ -1179,11 +1228,288 @@ def assert_rank(tensor, expected_rank, name=None):
...
+                    do_return_all_layers=False,
+                    low_memory_mode=config.low_memory_mode and is_training,
+                )
+                self.all_encoder_layers = None
+
+    def embedding_postprocessor(self, *args, **kwargs):
+        def table_pos_embed(
+            input_context, positional_channels, batch_size, seq_length, width
+        ):
+            max_row_col_embedding = 1024
```
Comment (madisonmay) [inline finetune/base_models/bert/modeling.py]: nit: could consider attaching this to config somewhere if we ever anticipate changing it

File: finetune/base_models/huggingface/models.py [hunk 4]
```diff
@@ -50,31 +67,39 @@
...
-        "low_memory_mode": True,
-        "batch_size": 2,
-        "n_epochs": 16,
-        "lr": 1.68e-3,
-        "max_grad_norm": 1.1,
-        "lr_warmup": 0.7,  # Not a typo - just weird.
-        "l2_reg": 0.194,
-    },
-)
+# FIXME: - Currently very broken.
```
Comment (madisonmay) [inline finetune/base_models/huggingface/models.py:70]: I don't think we need to fix this personally

File: finetune/datasets/reuters.py [hunk 5]
```diff
@@ -78,7 +79,7 @@ def download(self):
         test_size=0.2,
         random_state=42
     )
-    model = SequenceLabeler(batch_size=1, n_epochs=3, val_size=0.0, max_length=512, chunk_long_sequences=True, subtoken_predictions=False, crf_sequence_labeling=True, multi_label_sequences=False)
+    model = SequenceLabeler(batch_size=1, n_epochs=3, val_size=0.0, max_length=512, chunk_long_sequences=True, subtoken_predictions=False, crf_sequence_labeling=True, multi_label_sequences=False, base_model=HFDebertaV3Base)
```
Comment (madisonmay) [inline finetune/datasets/reuters.py]: nit: doesn't have to be committed

File: finetune/target_models/document_labeling.py [hunk 6]
```diff
@@ -3,72 +3,25 @@
...
-                    'bottom': pos["bottom"] * y_norm,
-                    'left': pos["left"] * x_norm,
-                    'right': pos["right"] * x_norm,
-                    'start': offset["start"],
-                    'end': offset["end"],
-                    'text': token["text"],
-                }
-	        )
-    return context
+    if base_model.get_context_fn is not None:
```
Comment (madisonmay) [inline finetune/target_models/document_labeling.py:14]: Nice -- I dig this refactor.

File: finetune/target_models/sequence_labeling.py [hunk 7]
```diff
@@ -496,6 +500,7 @@ def _predict_decode(
         doc_idx = -1
         doc_annotations = []
         raw_text = [data.get("raw_text", data["X"]) for data in zipped_data]
+        post_proc_time = 0.0
```
Comment (madisonmay) [inline finetune/target_models/sequence_labeling.py]: May / may not want to remove timings

File: finetune/util/huggingface_interface.py [hunk 8]
```diff
@@ -1,3 +1,5 @@
+from math import exp
+from multiprocessing.sharedctypes import Value
```
Comment (madisonmay) [inline finetune/util/huggingface_interface.py]: Unintended import

File: tests/test_base_models.py [hunk 9]
```diff
@@ -501,15 +505,17 @@ def test_featurize_sequence(self):
 
 
 def _test_featurize_sequence(self, model_fn):
-    test_input = ["""Pirateipsum: Corsair bilge rat interloper. Nipperkin
+    test_input = [
+        """Pirateipsum: Corsair bilge rat interloper. Nipperkin
```
Comment (madisonmay) [inline tests/test_base_models.py:509]: pirate ipsum :pirate_flag:

**Review by madisonmay — APPROVED** [review]

**Pull request merged**